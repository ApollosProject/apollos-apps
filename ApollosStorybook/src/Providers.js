import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
