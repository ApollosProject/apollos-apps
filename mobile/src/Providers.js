import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
