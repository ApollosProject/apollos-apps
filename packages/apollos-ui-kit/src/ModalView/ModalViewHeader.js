import React from 'react';
import { StatusBar, View } from 'react-native';
import Color from 'color';

import styled from '../styled';
import { ButtonIcon } from '../Button';
import { withTheme } from '../theme';

const StyledButtonIcon = withTheme(({ theme }) => ({
  fill: theme.colors.text.tertiary,
  size: theme.sizing.baseUnit,
  iconPadding: theme.sizing.baseUnit * 1.75,
}))(ButtonIcon);

const ButtonWrapper = styled(({ theme, right, left }) => ({
  position: 'absolute',
  top: theme.sizing.baseUnit * 1.5,
  right: right ? theme.sizing.baseUnit * 1.5 : null,
  left: left ? theme.sizing.baseUnit * 1.5 : null,
  width: theme.sizing.baseUnit * 1.75,
  height: theme.sizing.baseUnit * 1.75,
  borderRadius: theme.sizing.baseUnit * 1.75,
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: Color(theme.colors.text.primary).fade(0.1),
  zIndex: 2,
}))(View);

const Handle = styled({
  // helps in swipe-to-close gesture
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  height: 75,
})(View);

const ModalView = ({ onClose, onBack }) => (
  <>
    <StatusBar hidden />
    <Handle />
    {onBack ? (
      <ButtonWrapper left>
        <StyledButtonIcon name={'arrow-back'} onPress={onBack} />
      </ButtonWrapper>
    ) : null}
    {onClose ? (
      <ButtonWrapper right>
        <StyledButtonIcon name={'close'} onPress={onClose} />
      </ButtonWrapper>
    ) : null}
  </>
);

export default ModalView;
