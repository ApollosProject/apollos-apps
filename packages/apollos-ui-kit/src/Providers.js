import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
