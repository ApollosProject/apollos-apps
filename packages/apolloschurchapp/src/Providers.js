import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
