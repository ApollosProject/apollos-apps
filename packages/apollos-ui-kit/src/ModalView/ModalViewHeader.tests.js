import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../Providers';

import ModalViewHeader from './ModalViewHeader';

describe('The ModalView/ModalViewHeader component', () => {
  it('should render a back button', () => {
    const tree = renderer.create(
      <Providers>
        <ModalViewHeader onBack={jest.fn()} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a close button', () => {
    const tree = renderer.create(
      <Providers>
        <ModalViewHeader onClose={jest.fn()} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render both buttons', () => {
    const tree = renderer.create(
      <Providers>
        <ModalViewHeader onBack={jest.fn()} onClose={jest.fn()} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
