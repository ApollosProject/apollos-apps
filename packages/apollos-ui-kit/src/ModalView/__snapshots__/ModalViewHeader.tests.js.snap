// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ModalView/ModalViewHeader component should render a back button 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "height": 75,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      left={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": Object {
            "color": Array [
              48,
              48,
              48,
            ],
            "model": "rgb",
            "valpha": 0.9,
          },
          "borderRadius": 28,
          "height": 28,
          "justifyContent": "center",
          "left": 24,
          "position": "absolute",
          "right": null,
          "top": 24,
          "width": 28,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          iconPadding={28}
          style={
            Object {
              "padding": 28,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGPath
              d="M14.35 21L6.2 12.46c-.26-.28-.26-.64 0-.92L14.34 3l1.65.9L8.27 12 16 20.1l-1.65.9z"
              fill={
                Array [
                  0,
                  0.6470588235294118,
                  0.6470588235294118,
                  0.6470588235294118,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`The ModalView/ModalViewHeader component should render a close button 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "height": 75,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      right={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": Object {
            "color": Array [
              48,
              48,
              48,
            ],
            "model": "rgb",
            "valpha": 0.9,
          },
          "borderRadius": 28,
          "height": 28,
          "justifyContent": "center",
          "left": null,
          "position": "absolute",
          "right": 24,
          "top": 24,
          "width": 28,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          iconPadding={28}
          style={
            Object {
              "padding": 28,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGPath
              d="M13.07 12l7.2-7.2.5-.5c.3-.3.3-.78 0-1.08-.3-.3-.76-.3-1.06 0l-.5.5-7.2 7.2-7.2-7.2-.5-.5c-.3-.3-.77-.3-1.07 0-.3.3-.3.77 0 1.07l.5.5 7.2 7.2-7.7 7.7c-.3.3-.3.77 0 1.07.3.3.77.3 1.07 0l7.7-7.7 7.7 7.7c.3.3.78.3 1.08 0 .3-.3.3-.77 0-1.07l-7.7-7.7z"
              fill={
                Array [
                  0,
                  0.6470588235294118,
                  0.6470588235294118,
                  0.6470588235294118,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`The ModalView/ModalViewHeader component should render both buttons 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "height": 75,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      left={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": Object {
            "color": Array [
              48,
              48,
              48,
            ],
            "model": "rgb",
            "valpha": 0.9,
          },
          "borderRadius": 28,
          "height": 28,
          "justifyContent": "center",
          "left": 24,
          "position": "absolute",
          "right": null,
          "top": 24,
          "width": 28,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          iconPadding={28}
          style={
            Object {
              "padding": 28,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGPath
              d="M14.35 21L6.2 12.46c-.26-.28-.26-.64 0-.92L14.34 3l1.65.9L8.27 12 16 20.1l-1.65.9z"
              fill={
                Array [
                  0,
                  0.6470588235294118,
                  0.6470588235294118,
                  0.6470588235294118,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      right={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": Object {
            "color": Array [
              48,
              48,
              48,
            ],
            "model": "rgb",
            "valpha": 0.9,
          },
          "borderRadius": 28,
          "height": 28,
          "justifyContent": "center",
          "left": null,
          "position": "absolute",
          "right": 24,
          "top": 24,
          "width": 28,
          "zIndex": 2,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          iconPadding={28}
          style={
            Object {
              "padding": 28,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="16"
            bbWidth="16"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGPath
              d="M13.07 12l7.2-7.2.5-.5c.3-.3.3-.78 0-1.08-.3-.3-.76-.3-1.06 0l-.5.5-7.2 7.2-7.2-7.2-.5-.5c-.3-.3-.77-.3-1.07 0-.3.3-.3.77 0 1.07l.5.5 7.2 7.2-7.7 7.7c-.3.3-.3.77 0 1.07.3.3.77.3 1.07 0l7.7-7.7 7.7 7.7c.3.3.78.3 1.08 0 .3-.3.3-.77 0-1.07l-7.7-7.7z"
              fill={
                Array [
                  0,
                  0.6470588235294118,
                  0.6470588235294118,
                  0.6470588235294118,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
